### ISSUE TITLE
Fuzz statistics calculations

### ISSUE THREAD

**Issue by lwwmanning** [vortex-data/vortex#1307]
We should fuzz statistics by computing all possible stats on an array via `array.statistics.compute_all(all::<Stat>())` or some such, canonicalizing that array, and then calling `compute_statistics` on the canonicalized array (one by one). 

If the statistic is present in the first set, it should also be present and with equal value in the second set.

**Comment by gatesn** [comment]
Doesn't that break when you hard-code canonicalize to copy over all stats?

**Comment by lwwmanning** [comment]
Nah, because `compute_statistics` bypasses the cached values altogether

**Comment by gatesn** [comment]
These are being converted to compute functions so should fall under that umbrella of fuzzing
